trin-core/src/portalnet/discovery.rs

```diff
@@ -127,22 +123,47 @@ impl Discovery {
         })
     }
 
-    pub async fn start(&mut self) -> Result<(), String> {
+    pub async fn start(&mut self) -> Result<mpsc::Receiver<TalkRequest>, String> {
+        info!(
+            "Starting discv5 with local enr encoded={:?} decoded={}",
+            self.local_enr(),
+            self.local_enr()
+        );
+
         let _ = self
             .discv5
             .start(self.listen_socket)
             .await
             .map_err(|e| format!("Failed to start discv5 server: {:?}", e))?;
         self.started = true;
-        Ok(())
+
+        let mut event_rx = self.discv5.event_stream().await.unwrap();
+
+        let (talk_req_tx, talk_req_rx) = mpsc::channel(TALKREQ_CHANNEL_BUFFER);
+
+        tokio::spawn(async move {
+            while let Some(event) = event_rx.recv().await {
+                // Forward all TALKREQ messages.
+                if let Discv5Event::TalkRequest(talk_req) = event {
+                    let _ = talk_req_tx.send(talk_req).await;
```

Comment:
Do we even have to `await` here, since we're ignoring the result? Seems like we could do a more fire-and-forget type option, which might improve performance in some scenarios.

Reply:
Nevermind, I was thinking in python again.

> This method will never block the current thread.
https://doc.rust-lang.org/std/sync/mpsc/struct.Sender.html#method.send